Tests/Realm.Tests/Sync/SyncTestBase.cs

```diff
@@ -138,6 +138,13 @@ protected User GetFakeUser(App? app = null, string? id = null, string? refreshTo
             return new User(handle, app);
         }
 
+        // This could be useful when opening a "fake" sync realm locally
+        protected void SetFakeSyncRoute(App? app)
+        {
+            app ??= DefaultApp;
+            app.Handle.SetFakeSyncRouteForTesting();
```

Comment:
I'm not sure how I feel about this - I know it's just for tests, but if this is called with one of our integration apps, it would sets its sync route to a fake value, even though a real value should exist. I don't know if sync will refresh that route at some point, but I worry that as it stands, it's possible that running this test first, then running some integration tests that use the app, those will fail due to the route being misconfigured.

Reply:
Mmmh, I understand. Good news is that all the tests are passing now, including sync tests, so I suppose it's not a problem "yet".
If we want to be more robust for the tests we could:
- Set the fake value only if the sync route is currently empty/null
- Set it to null/empty again at the end of the test (if modified)
- Maybe create infrastructure in core to create those fake synced realm more "organically"?
I am not sure if it's worth to do this though.

Reply:
Given that this is the only holdup and it's only for tests, what do you think about merging this and creating a follow up ticket so we come back to it when the core investigation is finished?

Reply:
Sure